geo/aws.go

```diff
@@ -0,0 +1,172 @@
+package geo
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"log"
+	"net"
+	"os"
+	"path/filepath"
+	"sync"
+	"time"
+)
+
+const (
+	AWSGeoUrl  = "https://ip-ranges.amazonaws.com/ip-ranges.json"
+	AWSExpired = 7 * 24 * time.Hour
+	AWSFile    = "aws-ip.json"
+)
+
+type AWSProvider struct {
+	Info     *AWSGeoInfo
+	fileName string
+	url      string
+	expire   time.Duration
+	ready    bool
+	mux      sync.Mutex
```

Comment:
This lock cannot solve the problem because different tuna instance will have different lock